Xamarin.Forms.Core/Routing.cs

```diff
@@ -88,13 +104,7 @@ public static void SetRoute(Element obj, string value)
 		}
 
 		static bool ValidateRoute(string route)
-		{
-			// Honestly this could probably be expanded to allow any URI allowable character
-			// I just dont want to figure out what that validation looks like.
-			// It does however need to be lowercase since uri case sensitivity is a bit touchy
-			Regex r = new Regex(@"^[a-z|\/]*$");
-			return r.IsMatch(route);
-		}
+			=> new Regex(@"^[-a-zA-Z0-9@:%._\+~=\/]{1,100}$").IsMatch(route);
```

Comment:
Instead of having each RegisterRoute overload check the result of ValidateRoute and throw an exception, perhaps ValidateRoute should return void and throw an exception if the route is invalid.

Then ValidateRoute can throw detail exception messages.

Reply:
So we're thinking just change the route check to instantiate a URI 
if it succeeds it's valid if it doesn't then it's not valid. Instead of trying to use a REGEX